models/yang/sonic/sonic-bgp-clear.yang

```diff
@@ -0,0 +1,162 @@
+module sonic-bgp-clear {
+    namespace "http://github.com/Azure/sonic-bgp-clear";
+    prefix sbgpclear;
+    yang-version 1.1;
+
+    import ietf-inet-types {
+        prefix inet;
+    }
+
+    import ietf-yang-types {
+        prefix yang;
+    }
+
+    import sonic-port {
+        prefix prt;
+    }
+
+    import sonic-portchannel {
+        prefix spc;
+    }
+
+    import sonic-vlan {
+        prefix svlan;
+    }
+
+    import sonic-mgmt-port {
+        prefix smt;
+    }
+
+    organization
+        "SONiC";
+
+    contact
+        "SONiC";
+
+    description
+        "SONiC yang for BGP clear options";
+
+    revision 2020-12-01 {
+        description
+            "Initial revision.";
+    }
+
+   rpc clear-bgp {
+
+    description "RPC for clearing BGP. Input can be :
+        1. All BGP sessions 
+        2. All interfaces
+        3. All interfaces of a particular type";
+
+    input {
+        leaf clear-all {
+            type boolean;
+            description
+                "Clear all BGP peers.";
+        }
+
+        leaf vrf-name {
+            type string;
+            description
+                "Clear BGP by VRF name.";
+        }
+
+        leaf family {
+            type enumeration {
+                enum IPv4;
+                enum IPv6;
+            }
+            description "Address family (IPv4/IPv6)";
+        }
+
+        choice option {
+
+            leaf all {
+                type boolean;
+                description
+                    "Clear all BGP peers.";
+            }
+            leaf external {
+                type boolean;
+                description
+                    "Clear all BGP external peers.";
+            }
+
+            leaf address {
+                type inet:ip-address;
+                description
+                    "Clear BGP by neighbor address.";
+            }
+
+            leaf interface {
+                type union {
+                    type leafref {
+                        path "/prt:sonic-port/prt:PORT/prt:PORT_LIST/prt:ifname";
+                    }
+                    type leafref {
+                        path "/spc:sonic-portchannel/spc:PORTCHANNEL/spc:PORTCHANNEL_LIST/spc:name";
+                    }
+                    type leafref {
+                        path "/svlan:sonic-vlan/svlan:VLAN/svlan:VLAN_LIST/svlan:name";
+                    }
+                    type leafref {
+                        path "/smt:sonic-mgmt-port/smt:MGMT_PORT/smt:MGMT_PORT_LIST/smt:ifname";
+                    }
+                }
+                description
+                    "Interface name: Parameter for deleting all ARP  entries
```

Comment:
Not ARP entries.. please fix description.

Reply:
Done